### PR TITLE
Bump iOS SDK to v3.0.0

### PR DESCRIPTION
**Pull request by fotiDim**
Spotify released today [v3.0.0](https://github.com/spotify/ios-sdk/releases/tag/v3.0.0) of the iOS SDK and they removed the `checkIfSpotifyAppIsActive` method.
@brim-borium please verify if it will still be supported on Android.

### CODE REVIEW

File: ios/prepare-iOS-SDK.sh [hunk 1]
```diff
@@ -6,5 +6,5 @@ FRAMEWORK_NAME="SpotifyiOS.xcframework"
 rm -fR ${REPO_NAME}
 mkdir ${REPO_NAME}
 git clone https://github.com/spotify/${REPO_NAME}
-git -C ${REPO_NAME} checkout checkout tags/v2.1.7
+git -C ${REPO_NAME} checkout checkout tags/v3.0.0
```
Comment (haashem) [inline ios/prepare-iOS-SDK.sh:9]: there is duplicate checkout.

**Comment by brim-borium** [comment]
@fotiDim we currently don't use any method from the sdk itself but do something like this for android

`                .setResultCallback { playerState ->  result.success(!playerState.isPaused && playerState.playbackSpeed > 0)}
`

So this shoudl still work

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 3.0.1
```
Comment (brim-borium) [inline CHANGELOG.md:1]: I think this should at least be a minor change  or even a breaking change, what do you think?

Comment (fotiDim) [reply CHANGELOG.md:1]: Since our 3.0 was released only a few days ago I think it is OK to stick to it. I expect most people to not have upgraded already.

**Comment by fotiDim** [comment]
> > @fotiDim we currently don't use any method from the sdk itself but do something like this for android
> > ```
> > .setResultCallback { playerState ->  result.success(!playerState.isPaused && playerState.playbackSpeed > 0)}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So this should still work
> 
> Maybe this is something we can adapt for iOS as well ?

Good point. I will make a commit. Then this PR is not even a breaking change.

**Comment by fotiDim** [comment]
@brim-borium I have made the change as suggested. Let me know if you prefer to keep it or not, like I mention in my comment above.

**Comment by brim-borium** [comment]
> > > @fotiDim we currently don't use any method from the sdk itself but do something like this for android
> > > ```
> > > .setResultCallback { playerState ->  result.success(!playerState.isPaused && playerState.playbackSpeed > 0)}
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > So this should still work
> > 
> > 
> > Maybe this is something we can adapt for iOS as well ?
> 
> Good point. I will make a commit. Then this PR is not even a breaking change.
> 
> EDIT: On second thought, we [added the Android implementation](https://github.com/brim-borium/spotify_sdk/pull/108) only to have feature parity with iOS. Now that the native iOS method is gone maybe we should completely remove it from all platforms as it is custom business logic that by convention does not belong to the library. It could also be implemented by the end developer. I will make the commit anyway and if we decide against it I can remove it.

You are right the I will remove the android code on this branch as well and then we have feature parity to the sdks

**Comment by brim-borium** [comment]
@fotiDim I removed the android method and removed it from the sdk and the example as well. Can you remove it from iOS and after that we can merge this

**Comment by fotiDim** [comment]
@brim-borium done. Please double that all is all right.

**Comment by brim-borium** [comment]
> @brim-borium done. Please double that all is all right.

Will do

**Comment by brim-borium** [comment]
@fotiDim android looks good you can merge if iOS is fine as well :)

**Review by brim-borium — APPROVED** [review]

**Comment by fotiDim** [comment]
All good. Merging.

**Pull request merged**